links/src/commonTest/kotlin/com/splendo/kaluga/links/manager/DefaultLinksManagerTest.kt

```diff
@@ -20,23 +20,58 @@ package com.splendo.kaluga.links.manager
 import kotlinx.serialization.Serializable
 
 @Serializable
-data class Person(
-    val name: String,
-    val surname: String,
-    val spokenLanguages: List<Languages> = emptyList()
+enum class MyEnum {
+    A, B, C
+}
+
+@Serializable
+data class DataTypesValues(
+    val stringValue: String,
+    val intValue: Int,
+    val longValue: Long,
+    val floatValue: Float,
+    val doubleValue: Double,
+    val booleanValue: Boolean,
+    val enumValue: MyEnum,
+    val listValue: List<String>,
+    val nullableValue: String? = null
 ) {
     companion object {
-        val dummyUrl = "http://url.com?name=Corrado&surname=Quattrocchi&spokenLanguages=3&spokenLanguages=ITALIAN&spokenLanguages=ENGLISH&spokenLanguages=DUTCH"
-        val dummyPerson = Person(
-            "Corrado",
-            "Quattrocchi",
-            listOf(Languages.ITALIAN, Languages.ENGLISH, Languages.DUTCH)
+        val validParameters = mapOf(
+            "stringValue" to "Test",
+            "intValue" to 0,
+            "longValue" to 3L,
+            "floatValue" to 3.14f,
+            "doubleValue" to 3.14,
+            "booleanValue" to true,
+            "enumValue" to "A",
+            "listValue.0" to "zero",
+            "listValue.1" to "first",
+            "listValue.2" to "second",
+            "nullableValue" to null // can also be excluded from values since has default value.
+        )
+        val expectedValidValues = DataTypesValues(
+            "Test",
+            0,
+            3L,
+            3.14f,
+            3.14,
+            true,
+            MyEnum.A,
+            listOf("zero", "first", "second"),
+            null
         )
-    }
-}
 
-enum class Languages {
-    ITALIAN,
-    ENGLISH,
-    DUTCH,
+        val url = """https://test.io?
+                stringValue=Test&
+                intValue=0&
+                longValue=3&
+                floatValue=3.14&
+                doubleValue=3.14&
+                booleanValue=true&
+                enumValue=A&
+                listValue.0=zero&
+                listValue.1=first&
```

Comment:
How it works, if you have `nullableValue` inside URL without any value? Should it be set a true (value present)?

`https://test.io/intValue=0&nullableValue&doubleValue=3.14`

So nullable parameter is present without any value.

Reply:
In case of ios `NSURLQueryItem.value` returns null while in case of android an empty string.
I could fix it by excluding the value if is set as "" in `extractQuery` method, but that means that every value that is a empty string will be set as null

Reply:
Like this 759b37a